packages/discord.js/typings/index.d.ts

```diff
@@ -473,43 +473,37 @@ export class ButtonInteraction<Cached extends CacheType = CacheType> extends Mes
   public inRawGuild(): this is ButtonInteraction<'raw'>;
 }
 
-export type KeyedEnum<K, T> = {
-  [Key in keyof K]: T | string;
-};
-
-export type EnumValueMapped<E extends KeyedEnum<T, number>, T extends Partial<Record<keyof E, unknown>>> = T & {
-  [Key in keyof T as E[Key]]: T[Key];
-};
-
-export type MappedChannelCategoryTypes = EnumValueMapped<
-  typeof ChannelType,
-  {
-    GuildNews: NewsChannel;
-    GuildVoice: VoiceChannel;
-    GuildText: TextChannel;
-    GuildStore: StoreChannel;
-    GuildStageVoice: StageChannel;
-  }
->;
+export interface MappedChannelCategoryTypes {
+  [ChannelType.GuildNews]: NewsChannel;
+  [ChannelType.GuildVoice]: VoiceChannel;
+  [ChannelType.GuildText]: TextChannel;
+  [ChannelType.GuildStore]: StoreChannel;
+  [ChannelType.GuildStageVoice]: StageChannel;
+}
 
-export type CategoryChannelTypes = ExcludeEnum<
-  typeof ChannelType,
-  'DM' | 'GroupDM' | 'GuildPublicThread' | 'GuildNewsThread' | 'GuildPrivateThread' | 'GuildCategory'
+export type CategoryChannelType = Exclude<
+  ChannelType,
+  | ChannelType.DM
+  | ChannelType.GroupDM
+  | ChannelType.GuildPublicThread
+  | ChannelType.GuildNewsThread
+  | ChannelType.GuildPrivateThread
+  | ChannelType.GuildCategory
 >;
 
 export class CategoryChannel extends GuildChannel {
   public readonly children: Collection<Snowflake, Exclude<NonThreadGuildBasedChannel, CategoryChannel>>;
-  public type: 'GuildCategory';
+  public type: ChannelType.GuildCategory;
```

Comment:
Does this actually enforce it to the value? I know in TS we had to cast as const

Reply:
Yeah it complains if it isn't an enum value.